File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.java
class Solution {
    public int[] minBitwiseArray(List<Integer> nums) {
        int n = nums.size();
        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            int target = nums.get(i);
            int value = linearSearch(target);
            ans[i] = value;
        }

        return ans;
    }

    static int linearSearch(int target) {
        for (int i = 1; i <= target; i++) {
            if (((i)|(i+1)) ==  target) {
                return i;
            }
        }
        return -1;
    }
}

File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans;

        for (int i=0; i<n; i++){
            if (nums[i] == 2){
                ans.push_back(-1);
                continue;
            }

            int l = 0;
            int copy = nums[i];
            while (copy & 1 == 1){
                l++;
                copy >>= 1;
            }
            l = l-1;
            int mask = 1 << l;
            ans.push_back(nums[i]^mask);
        }

        return ans;
    }
};